tox.ini

```diff
@@ -39,82 +40,63 @@ commands =
 description = Check code against coding style standards
 deps =
     black
+    codespell
     flake8
     flake8-docstrings
     flake8-copyright
     flake8-builtins
     pyproject-flake8
     pep8-naming
     isort
-    codespell
 commands =
+    codespell {[vars]lib_path}
+    codespell . --skip .git --skip .tox --skip build --skip lib --skip venv --skip .mypy_cache
     # pflake8 wrapper supports config from pyproject.toml
     pflake8 {[vars]all_path}
     isort --check-only --diff {[vars]all_path}
     black --check --diff {[vars]all_path}
-    codespell {[vars]lib_path}
-    codespell . --skip .git --skip .tox --skip build --skip lib --skip venv --skip .mypy_cache
 
-[testenv:static]
+[testenv:static-{charm,lib,unit,integration}]
 description = Run static analysis checks
+setenv =
+    unit: MYPYPATH = {[vars]tst_path}/unit
+    integration: MYPYPATH = {[vars]tst_path}/integration
 deps =
-    -r{toxinidir}/requirements.txt
     mypy
+    types-dataclasses
     types-PyYAML
-    pytest
-    pytest-operator
-    juju
-    types-setuptools
-    types-toml
     types-requests
+    charm: -r{toxinidir}/requirements.txt
+    lib: git+https://github.com/canonical/operator#egg=ops
```

Comment:
This is a bit risky because requirements.txt is from pypi. The two should probably match.

Reply:
This also matches the other repos (alert manager, prometheus, etc). If we want to fix it, let's do a unified change later.